src/components/dialog_preference/index.tsx

```diff
@@ -32,10 +67,34 @@ const PreferenceDialog: FC = () => {
           {`Any changes will be automatically saved to the cloud`}
         </DialogContentText>
         <FormGroup>
-          <FormControlLabel control={<Checkbox />} label="English" />
-          <FormControlLabel control={<Checkbox />} label="Chinese" />
-          <FormControlLabel control={<Checkbox />} label="Japanese" />
-          <FormControlLabel control={<Checkbox />} label="Korean" />
+          <FormControlLabel
+            checked={preference?.nativeLanguages.includes(`en`)}
+            control={<Checkbox />}
+            label="English"
+            onChange={onChange}
+            value={privateSupportedLanguages.English}
+          />
+          <FormControlLabel
+            checked={preference?.nativeLanguages.includes(`zh`)}
+            control={<Checkbox />}
+            label="Chinese"
+            onChange={onChange}
+            value={privateSupportedLanguages.Chinese}
+          />
+          <FormControlLabel
+            checked={preference?.nativeLanguages.includes(`ja`)}
+            control={<Checkbox />}
+            label="Japanese"
+            onChange={onChange}
+            value={privateSupportedLanguages.Japanese}
+          />
+          <FormControlLabel
+            checked={preference?.nativeLanguages.includes(`ko`)}
+            control={<Checkbox />}
+            label="Korean"
+            onChange={onChange}
+            value={privateSupportedLanguages.Korean}
+          />
```

Comment:
Do you think you can create an atom & refactor?

Reply:
Handled in [fb70732](https://github.com/ajktown/wordnote/pull/101/commits/fb707327d34b936da2a90007b91c1b35851c4f1a)